src/yaml/from_yaml.cr

```diff
@@ -22,39 +22,87 @@ end
 
 def Nil.new(pull : YAML::PullParser)
   location = pull.location
-  value = pull.read_scalar
-  if value.empty?
+  value = pull.read_plain_scalar
+  if YAML::NULL_VALUES.includes? value
     nil
   else
-    raise YAML::ParseException.new("Expected nil, not #{value}", *location)
+    raise YAML::ParseException.new("Expected null, not '#{value}'", *location)
   end
 end
 
 def Bool.new(pull : YAML::PullParser)
-  pull.read_scalar == "true"
+  location = pull.location
+  value = pull.read_plain_scalar
+  if YAML::TRUE_VALUES.includes? value
+    true
+  elsif YAML::FALSE_VALUES.includes? value
+    false
+  else
+    raise YAML::ParseException.new("Expected boolean, not '#{value}'", *location)
+  end
 end
 
 {% for type in %w(Int8 Int16 Int32 Int64 UInt8 UInt16 UInt32 UInt64) %}
```

Comment:
Maybe use the bit size instead of IntXX constants, and avoid raising/rescuing an exception, just to raise again:

```crystal
{% for bits in [8, 16, 32, 64] %}
  {% for type in %w(Int UInt) %}
    def {{type.id}}{{bits}}
      location = pull.location
      value = pull.read_plain_scalar

      number = {% if type == "UInt" %}
        value.to_u{{bits}}?(underscore: false)
      {% else %}
        value.to_i{{bits}}?(underscore: false)
      {% end %}
      return number if number

      raise YAML::ParseException.new("Invalid {{type.id}}{{bits}} number", *location)
    end
  {% end %}
{% end %}
```